### DISCUSSION TITLE
👨‍💻☎ WinUI Community Call: March 15, 2023

### DISCUSSION THREAD

**Discussion by gabbybilka** [microsoft/microsoft-ui-xaml#8187] [Announcements]
### Due to a technical issue we needed to reupload the video recording, it is available here: https://aka.ms/winuicall-03-15-23

# Details
Date: March 15, 2023
Time: 16:00-17:00 UTC (9:00-10:00am Pacific)

Anyone and everyone is welcome - no pre-registration is required.
This will be an informal interactive live stream directly with members of our engineering team.

# Format
The community call is a call among the WinUI team that is live-streamed onto YouTube. We present on new updates, share information, welcome guests, and answer your questions.

### **Agenda**
- Intro
- WinUI Gallery updates
- Technical challenges developing the WinAppSDK File Explorer
- Latest news regarding WinAppSDK & WinUI
- Q&A

### **Q&A Code of Conduct**
- To help ensure Q&A remains beneficial for the majority of the call's developer audience, please consider whether your question could be framed so that it might be more broadly relevant to our larger developer audience by ensuring it is reasonably concise, broadly beneficial, well-suited for the fast-paced live Q&A format, or able to spark open ended discussion.  
- Please understand some questions are not well-suited for the fast-paced live Q&A format and should be filed as standalone [Q&A](https://github.com/microsoft/microsoft-ui-xaml/discussions/new?category=q-a&title=%5BYour+Question+Here%5D) discussions where they can be most appropriately handled. Excluding presentation topic requests, this generally includes requests for dedicated follow-ups or technically complex questions that require investigative time or expertise not included in this month's guest list.

### Leave us your questions in the comments on this issue or live in the YouTube chat during the stream!

**Comment by eduardobragaxz** [comment]
It would be great to have someone like Savoy again, to fish the questions, rather than leave it up to people to fish them themselves.

**Reply by MEK3DK** [reply]
Yes the February call ended in chaos, but hey it is WinUI 3 ;-)

**Comment by JanRajnoha** [comment]
Are you planning work (solve or close unrelated) on backlog? According to GitHub Insights, backlog is getting bigger and bigger and for people it is reason, why they are not using Win App SDK.
![image](https://user-images.githubusercontent.com/16069721/223120728-812acc0c-8304-4b74-be42-89346929c6c3.png)

**Reply by sotanakamura** [reply]
I think Almost all of issues are problems on programming, not bugs.

**Reply by JanRajnoha** [reply]
It doesn't matter. The problem is, there are too many issues, where some serious problems could be overseen. And how I wrote in original comment, this is indication for many devs why they are not using it -> too much issues looks suspicious and could look like dead project.

**Reply by sotanakamura** [reply]
I agree with you. What I want to say is there are many issues to close easily. Closing these issues helps WinUI 3 team to focus on serious bugs and WinUI 3 project to look active.

**Reply by JanRajnoha** [reply]
Oh, yeah, I agree. It missing option to close issue by users

**Comment by jschwizer99** [comment]
What is the process at MS for releasing features with a **preview flag**?

The tree view SelectionChanged() event feature has been merged for over 2 years https://github.com/microsoft/microsoft-ui-xaml/pull/2063. It was marked with a preview flag. But it seems currently not accessible in the code (1.2.4 and 1.3.0 preview1). I would expect that at least in a preview-marked version it should be accessible. See also https://github.com/microsoft/WindowsAppSDK/issues/2690 and https://github.com/microsoft/microsoft-ui-xaml/issues/6585.

**Comment by Diegorro98** [comment]
What is the standard to save sensitive data in WindowsAppSDK/UWP ? I'm building an open source application that stores user tokens that should not be accesible. At the moment I'm using DataProtectionProvider to generate a buffer that latter will be transformed to a string using base64 encoding and stored in settings using CommunityToolkit.WinUI.Helpers.ApplicationDataStorageHelper (Save and read methods).

Is this secure enough to avoid easy or not so easy readings? Or there is another ways to achieve certain level of sensitive data protection?

**Reply by omikhailov** [reply]
For UWP it is [PasswordVault ](https://learn.microsoft.com/en-us/uwp/api/windows.security.credentials.passwordvault) and encryption

**Reply by terrycox** [reply]
You might look at Doppler

**Comment by AathifMahir** [comment]
Are there any plans for improvement to XAML in General, Xaml has always been the same from the beginning. Regardless of Modern Techniques and etc..

For Example:

- Custom Xaml Binding Like Maui
- Improvements for Compiled Bindings
Etc..

**Reply by ghost** [reply]
Oh how I miss the ease-of-use of WPF Style Triggers! If that could be revived, it would be so beneficial. And using the VisualStateManager, or even the new way of handling control states and colors for hover, mousedown, etc. using Resources is just so much more verbose and dirty. Everything was so much easier in WPF.

**Comment by lhak** [comment]
Any update on https://github.com/microsoft/microsoft-ui-xaml/issues/3665? This issue basically blocks my application from using winui 3 (still having to use the now unsupported .net core 3 and XAML islands) and probably impacts all applications using win2d by leaking gigabytes of gpu memory.

**Reply by DominicMaas** [reply]
hmmm, I'm tracking down an issue with customers stating that an app I'm developing can use gigabytes of GPU memory (like 16GB) after running for a few days (I just reprod today it with about 24 hours of running)

Although not the same as your issue, I'm not using Win2D directly (just via Shadows I think), and the user cannot resize the window. 

Need to keep this in mind while I keep digging to find the issue.....

EDIT: I think this is unlikely the same issue. Might actually have to do with the `RotatorTile` control, or something I'm doing wrong. Need to wait and see

**Comment by Diegorro98** [comment]
Has the use of SVG images as app icons in the Start Menu and Taskbar been discussed internally at Microsoft? What are your thoughts?

**Comment by koenvd** [comment]
Can we get an update again on https://github.com/MicrosoftEdge/WebView2Feedback/issues/3003 (aka WinUI3 WebView2 mouse stop working while keyboard is still working)? Much appreciated!

**Comment by JanRajnoha** [comment]
Any new info on agenda? @gabbybilka

**Reply by gabbybilka** [reply]
Updated, see above.

**Comment by Ccarmichael92** [comment]
Any short term plan to decouple webview2 from windows app sdk?

**Comment by MarkBrainfire** [comment]
I'm looking forward to a great meeting!

![Poster](https://user-images.githubusercontent.com/78389339/225328330-c92faedf-175d-43e0-aac7-3fff997f2ead.jpg)

**Comment by kmgallahan** [comment]
1. **Do you foresee WASDK / WinUI 3 ever updating to use the XAML 2009 spec like MAUI?**

2. **Is anyone still working on Win2D? Plans?**

3. **Any plans to make improvements to the Visual Studio WinUI development experience? (literally anything at all in the next ~year)**

4. **In WinUI project files, are the automatically created entries referencing XAML files actually used for anything?** Removing `<None Remove="*.xaml/>` & `<Page Update="*.xaml" ... />` doesn't seem to change the build process at all.

5. Most (all?) of the code in the public WinUI repo references Windows.UI.Xaml namespace (WinUI 2). **Does that imply all the WinUI 3 code resides in a private parallel repo?** If yes, will that code be made public (at the WinUI 3 counterparts for what is already public from WinUI 2). Are there major differences besides the namespaces?

**Comment by kmgallahan** [comment]
Would it be possible to add support for the `XmlnsDefinition` & `XmlnsPrefix` assembly attributes to improve XAML namespace declarations?

Or otherwise some form of Global Usings but for XAML namespaces.

**Reply by AathifMahir** [reply]
Xaml been due on major improvements for years now, hope they'd start working on improvements to xaml very soon

**Comment by JanRajnoha** [comment]
Will be Community Call this month? @gabbybilka

**Reply by gabbybilka** [reply]
Yes, on the 19th. The discussion for questions to be answered during Live Q&A will be posted soon.
